mm2src/mm2_main/src/lp_swap/maker_swap.rs

```diff
@@ -678,10 +681,14 @@ impl MakerSwap {
         drop(send_abort_handle);
         let mut swap_events = vec![];
         if let Some(instructions) = payload.instructions() {
-            match self
-                .maker_coin
-                .validate_instructions(instructions, &self.secret_hash(), self.maker_amount.clone())
-            {
+            let maker_lock_duration =
+                (self.r().data.lock_duration as f64 * self.taker_coin.maker_locktime_multiplier()).ceil() as u64;
```

Comment:
> Can we use MakerSwapData::maker_payment_lock here?

No, it can't be used, we want the duration here not the timestamp. We can use `maker_payment_lock - started_at` but I don't think it's intuitive. What do you think?

> Also I've noticed that we call maker_locktime_multiplier on the taker coin in maker_swap.rs and taker_swap.rs. Is this expected?

Yes, this is expected. We use 1.5x multiplier only when lightning coin is taker. Maybe I should refactor this and use the multiplier based on the taker locktime duration, if it exceeds 24 hours we use 1.5x, otherwise we use 2x.
What do you think @sergeyboyko0791 ?

Reply:
Sure! Agree with that
> No, it can't be used, we want the duration here not the timestamp. We can use maker_payment_lock - started_at but I don't think it's intuitive. What do you think?

Reply:
In my opinion, it's not a common solution, because there can be coins in the future that will require more than 2x locktime multiplier. Right now I don't have any ideas 😁 
> Yes, this is expected. We use 1.5x multiplier only when lightning coin is taker. Maybe I should refactor this and use the multiplier based on the taker locktime duration, if it exceeds 24 hours we use 1.5x, otherwise we use 2x.

Reply:
For more context about this, @artemii235 proposed using 1.5x here https://github.com/KomodoPlatform/atomicDEX-API/issues/1045#issuecomment-1311628383, because taker locktime for lightning is too long and using 2x will be too much for the maker. So 1.5x is used because of the duration not the coin. https://github.com/KomodoPlatform/atomicDEX-API/blob/2a2a77ce647a9776c0f3d6b7cbc814db6b3e5423/mm2src/mm2_main/src/lp_swap.rs#L565-L567

Reply:
The fact that the locktime is chosen based on the taker coin configuration may confuse. But I have nothing to suggest right now, so let's resolve this conversation for now :)